plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/model/ComplexNode.java

```diff
@@ -112,6 +113,19 @@ public void addNode(Node node) {
 		nodeMap.put(node.name, node);
 	}
 
+	/** Removes a node from this instance */
+	public void removeNode(Node node) {
+		removeNodeChecks(node);
+		nodeMap.remove(node.name);
+	}
+
+	/** Checks invoked before a node from this instance is removed */
+	public void removeNodeChecks(Node node) {
+		assert entry != node : "FlowGraph malformed: Node not child of complex node";
```

Comment:
better if-throw,
or guava reconditions.
http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Preconditions.html#checkState(boolean)

This shouldn't depend on `-ea` user can enable or disable.

Reply:
Todo in GH-235